src/xc_integrator/local_work_driver/device/cuda/kernels/uvvars.cu

```diff
@@ -145,6 +146,31 @@ __global__ void eval_uvars_gga_kernel( size_t           ntasks,
   }
 }
 
+__global__ void eval_uvars_lda_uks_kernel( size_t        ntasks,
+                                       XCDeviceTask* tasks_device ) {
+
+  const int batch_idx = blockIdx.z;
+  if( batch_idx >= ntasks ) return;
+
+  auto& task = tasks_device[ batch_idx ];
+
+  const auto npts            = task.npts;
+  const auto nbf             = task.bfn_screening.nbe;
+
+  auto* den_pos_eval_device   = task.den_pos;
+  auto* den_neg_eval_device   = task.den_neg;
+
+
+  const int tid_y = blockIdx.y * blockDim.y + threadIdx.y;
+
+
+  if( tid_y < npts ) {
+    den_pos_eval_device[ tid_y ] += den_neg_eval_device[ tid_y ];
+    den_neg_eval_device[ tid_y ] *= -2.0;
```

Comment:
?

Reply:
This is a trick to go from (S,Z) -> (S+Z, S-Z) without using scratch space. I'll add a comment on the next pass

Reply:
Oh, don't rely on FP tricks like that. Compiler optimization can make a mess of them. Also, GPU's are _**really**_ dumb, what you have written will emit excessive global reads. This should be implemented ala
```
/// This allows for the reads/writes to be vectorized and only hit global once on each direction
const auto ps = den_p[...];
const auto pz = den_m[...];
den_p[...] = ps + pz;
den_m[...] = ps - pz; 
```